client/helpers.go

```diff
@@ -61,3 +63,11 @@ func IgnoreAccessDenied(err error) bool {
 
 	return errors.As(err, &detailedError) && detailedError.StatusCode == http.StatusForbidden
 }
+
+func IgnoreSubscriptionNotRegistered(err error) bool {
+	var azureErr *azure.RequestError
+	if errors.As(err, &azureErr) {
+		return azureErr.ServiceError != nil && azureErr.ServiceError.Code == "SubscriptionNotRegistered"
+	}
+	return false
+}
```

Comment:
Why not add this to the generic error classifier that way all resources can benefit from it?

Reply:
moved to the generic error classifier and fixed unreachable code there.